src/main/scala/com/microsoft/azure/cosmosdb/spark/rdd/CosmosDBRDDIterator.scala

```diff
@@ -197,12 +199,25 @@ class CosmosDBRDDIterator(hadoopConfig: mutable.Map[String, String],
         .getOrElse(FilterConverter.createQueryString(requiredColumns, filters))
       logInfo(s"CosmosDBRDDIterator::LazyReader, created query string: $queryString")
 
-      if (queryString == FilterConverter.defaultQuery) {
-        // If there is no filters, read feed should be used
-        connection.readDocuments(feedOpts)
-      } else {
-        connection.queryDocuments(queryString, feedOpts)
+      var iteratorDocument: Iterator[Document] = Iterator()
+
+      try {
+        if (queryString == FilterConverter.defaultQuery) {
+          // If there is no filters, read feed should be used
+          iteratorDocument = connection.readDocuments(feedOpts)
+        } else {
+          iteratorDocument = connection.queryDocuments(queryString, feedOpts)
```

Comment:
this code is shared between streaming and batch .. the handling you currently do is only for streaming, can we move this to only be used for streaming ?